plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/animator/CameraAnimator.kt

```diff
@@ -93,6 +97,55 @@ abstract class CameraAnimator<out T> (
     }
   }
 
+  /**
+   * The most recent value calculated by this <code>ValueAnimator</code> when there is just one
+   * property being animated. This value is only sensible while the animation is running. The main
+   * purpose for this read-only property is to retrieve the value from the <code>ValueAnimator</code>
+   * during a call to {@link AnimatorUpdateListener#onAnimationUpdate(ValueAnimator)}, which
+   * is called during each animation frame, immediately after the value is calculated.
+   *
+   * @return animatedValue The value most recently calculated by this <code>ValueAnimator</code> for
+   * the single property being animated. If there are several properties being animated
+   * (specified by several PropertyValuesHolder objects in the constructor), this function
+   * returns the animated value for the first of those objects.
+   */
+  override fun getAnimatedValue(): Any {
+    // For immediate animations on API 23, as we introduced the bypass logic in handleImmediateAnimationOnAPI23(),
+    // the returned ValueAnimator.getAnimatedValue() will be null, in this case, we should return the
+    // last configured target value, so we immediately jump to the target value.
+    return super.getAnimatedValue() ?: targets.last() as Any
+  }
+
+  /**
+   * Handle immediate animation(when duration and startDelay of the animation is 0) on devices running
+   * API 23 or below, by sending updates to animator listeners directly without triggering [ValueAnimator]'s start().
+   *
+   * @return true if the animation is handled, false otherwise
+   */
+  private fun handleImmediateAnimationOnAPI23(): Boolean {
+    // Devices with API <= 23 animating with duration = 0 will send initial value from Looper with some small delay as a result.
+    // Hence multiple immediate animations sent close to each other may cancel previous ones.
+    // For these cases, we bypass the ValueAnimator and emit the user registered AnimatorListeners and AnimatorUpdateListeners immediately.
+    if (Build.VERSION.SDK_INT <= 23) {
+      if (duration == 0L && startDelay == 0L) {
+        val tmpListeners = listeners.toList()
```

Comment:
Why is `toList` needed btw? To avoid ConcurrentModificationException? I guess it's still possible, if list is being changed from the different thread, `toList` also iterates the list. 
We probable need to make listeners CopyOnWriteArraySet to avoid any problems

Reply:
sounds good!

Reply:
I've changed our declared variables to CopyOnWriteArraySet, but kept the tmpListeners since it's from `ValueAnimator.getListeners()`, and it's handled in the same way in ValueAnimator itself: https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/animation/ValueAnimator.java#1329

Reply:
Indeed... Might be a problem, though will happen only with 23 API and animations with 0 lenght, probably won't happen that often 😆

Reply:
> avoid ConcurrentModificationException?

I encountered ConcurrentModificationException on my first try to run this fix 🙈